syntax/switch.go

```diff
@@ -83,24 +108,46 @@ func (c *CaseClause) Equal(d *CaseClause) bool {
 // the first argument will be false, and the Env must be ignored.
 type switchCont func(bool, *values.Env) (values.T, error)
 
-func (s *evalSwitch) evalCases(cs []*CaseClause) (values.T, error) {
+// newSwitcher returns a new switcher to evaluate a switch expression.
+func newSwitcher(sess *Session, env *values.Env, ident string, e *Expr, v values.T, t *types.T) *switcher {
+	return &switcher{
+		sess:  sess,
+		env:   env,
+		ident: ident,
+		e:     e,
+		v:     v,
+		t:     t,
+	}
+}
+
+// eval kicks off evaluation of the switch expression managed by s.
+func (s *switcher) eval() (values.T, error) {
+	return s.evalCases(s.e.CaseClauses)
+}
+
+func (s *switcher) evalCases(cs []*CaseClause) (values.T, error) {
 	if len(cs) == 0 {
-		// This means no case clause matched.  This is an error, as the switch
-		// expression has no value.
-		return nil, fmt.Errorf("%s: no case pattern matches value", s.pos)
+		// Exhaustiveness-checking should prevent us from getting here.
```

Comment:
in this case should we panic?

Reply:
Done in [separate PR](https://github.com/grailbio/reflow/pull/123).